ext/bcmath/bcmath.c

```diff
@@ -152,7 +175,11 @@ PHP_FUNCTION(bcadd)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (ZEND_NUM_ARGS() == 3) {
-		scale = (int) (scale_param < 0 ? 0 : scale_param);
+		if (scale_param < 0 || scale_param > INT_MAX) {
+			zend_argument_value_error(3, "must be between 0 and %d", INT_MAX);
+			RETURN_THROWS();
+		}
+		scale = (int) scale_param;
```

Comment:
We should drop the explicit `(int)` casts from `int scale = (int)BCG(bc_precision)`, as `bc_precision` is now an int. Here and elsewhere.

Reply:
`scale_param` is a `zend_long`, though.

Reply:
Sorry, I meant the line further up. Can't comment on it in the PR...

Reply:
Ah, indeed. Thanks!

Reply:
Done